src/cc_toolchain/cc_toolchain_config.bzl

```diff
@@ -0,0 +1,791 @@
+        flag_sets = [
+            flag_set(
+                actions = [
+                    ACTION_NAMES.c_compile,
+                    ACTION_NAMES.cpp_compile,
+                ],
+                flag_groups = [
+                    flag_group(
+                        flags = [
+                            "-no-canonical-prefixes",
+                        ] + ctx.attr.extra_no_canonical_prefixes_flags,
+                    ),
+                ],
+            ),
+        ],
+    )
+
+    result["no_canonical_system_headers"] = feature(
+        name = "no-canonical-system-headers",
+        flag_sets = [
+            flag_set(
+                actions = [
+                    ACTION_NAMES.c_compile,
+                    ACTION_NAMES.cpp_compile,
+                ],
+                flag_groups = [
+                    flag_group(
+                        flags = [
+                            "-fno-canonical-system-headers",
+                        ],
+                    ),
+                ],
+            ),
+        ],
+    )
+
+    result["has_configured_linker_path_feature"] = feature(name = "has_configured_linker_path")
+
+    result["copy_dynamic_libraries_to_binary_feature"] = \
+        feature(name = "copy_dynamic_libraries_to_binary")
+
+    result["user_link_flags_feature"] = feature(
+        name = "user_link_flags",
+        flag_sets = [
+            flag_set(
+                actions = ALL_LINK_ACTIONS,
+                flag_groups = [
+                    flag_group(
+                        flags = ["%{user_link_flags}"],
+                        iterate_over = "user_link_flags",
+                        expand_if_available = "user_link_flags",
+                    ),
+                ],
+            ),
+        ],
+    )
+
+    result["cpp17_feature"] = feature(
+        name = "c++17",
+        flag_sets = [
+            flag_set(
+                actions = ALL_CPP_ACTIONS,
+                flag_groups = [flag_group(flags = ["-std=c++17"])],
+            ),
+        ],
+    )
+
+    result["no-rtti"] = feature(
+        name = "no-rtti",
+        flag_sets = [
+            flag_set(
+                actions = ALL_CPP_ACTIONS,
+                flag_groups = [flag_group(flags = ["-fno-rtti"])],
+            ),
+        ],
+    )
+
+    result["no-exceptions"] = feature(
+        name = "no-exceptions",
+        flag_sets = [
+            flag_set(
+                actions = ALL_CPP_ACTIONS,
+                flag_groups = [flag_group(flags = ["-fno-exceptions"])],
+            ),
+        ],
+    )
+
+    result["no-unused-command-line-argument"] = feature(
+        name = "no-unused-command-line-argument",
+        flag_sets = [
+            flag_set(
+                actions = ALL_COMPILE_ACTIONS + ALL_LINK_ACTIONS,
+                flag_groups = [flag_group(flags = ["-Wno-unused-command-line-argument"])],
+            ),
+        ],
+    )
+
+    return result
+
+def _clang_impl(ctx):
```

Comment:
what's the difference between this and _stm32_impl?

Reply:
AFAIK clang will be used for our AI (k8 architecture), gcc wil be used for the stm32 firmware

Reply:
yup, this basically "wraps" the clang compiler, which is used for the `k8` CPU (ie the x86_64 architecture), gcc (or rather `arm-none-eabi-gcc` is used for compiling for the `stm32*` CPU's (really all ARM CPU's).

Reply:
thanks